src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java

```diff
@@ -198,6 +224,126 @@ Consumer<Message<UUID>> consumeCaseImportStart() {
+                    .build();
+            ResponseInputStream<GetObjectResponse> s3Object = s3Client.getObject(getRequest);
+            String fileName = s3Object.response().metadata().get(METADATA_FILE_NAME);
+            if (fileName == null) {
+                fileName = s3Key.substring(s3Key.lastIndexOf("/") + 1);
+            }
+            Long fileLength = s3Object.response().contentLength();
+            HttpHeaders headers = new HttpHeaders();
+            headers.setContentDisposition(ContentDisposition.builder("attachment").filename(fileName).build());
+            headers.setContentLength(fileLength);
+            InputStreamResource resource = new InputStreamResource(s3Object);
+            return ResponseEntity.ok()
+                    .headers(headers)
+                    .contentType(MediaType.APPLICATION_OCTET_STREAM)
+                    .body(resource);
+        } catch (NoSuchKeyException e) {
+            return ResponseEntity.notFound().build();
+        }
+    }
+
+    @Bean
+    Consumer<Message<UUID>> consumeCaseExportStart() {
+        return message -> {
+            UUID caseUuid = message.getPayload();
+            String format = message.getHeaders().get(NotificationService.HEADER_FORMAT, String.class);
+            String fileName = message.getHeaders().get(NotificationService.HEADER_FILE_NAME, String.class);
+            String userId = message.getHeaders().get(NotificationService.HEADER_USER_ID, String.class);
+            UUID exportUuid = extractExportUuid(message);
+            Map<String, Object> formatParameters = extractFormatParameters(message);
+            try {
+                LOGGER.debug("Processing export for case {} with format {}...", caseUuid, format);
+                ExportNetworkInfos exportNetworkInfos = networkConversionObserver.observeExportProcessing(
+                        format,
+                        () -> exportCase(caseUuid, format, fileName, formatParameters)
+                );
+                String s3Key = exportRootPath + "/" + exportUuid.toString() + "/" + fileName;
+                uploadFile(exportNetworkInfos.getTempFilePath(), s3Key, exportNetworkInfos.getNetworkName());
+                notificationService.emitCaseExportFinished(exportUuid, userId, null);
+            } catch (Exception e) {
+                String errorMsg = String.format("Export failed for case %s: %s", caseUuid, e.getMessage());
+                LOGGER.error(errorMsg);
+                notificationService.emitCaseExportFinished(exportUuid, userId, errorMsg);
+            }
+        };
+    }
+
+    private UUID extractExportUuid(Message<UUID> message) {
+        return message.getHeaders().get(NotificationService.HEADER_EXPORT_UUID, UUID.class);
+    }
+
+    private Map<String, Object> extractFormatParameters(Message<UUID> message) {
+        Map<String, Object> rawParameters = (Map<String, Object>) message.getHeaders().get(NotificationService.HEADER_EXPORT_PARAMETERS);
```

Comment:
Keep the comment that was in this code : 
// String longer than 1024 bytes are converted to com.rabbitmq.client.LongString (https://docs.spring.io/spring-amqp/docs/3.0.0/reference/html/#message-properties-converters)